api/valor_api/backend/core/llm_clients.py

```diff
@@ -0,0 +1,521 @@
+
+        Returns
+        -------
+        Any
+            The messages formatted for the API. By default, the messages are left in the OpenAI format.
+        """
+        return messages
+
+    def __call__(
+        self,
+        messages: list[dict[str, str]],
+    ) -> Any:
+        """
+        Call to the API. Not implemented for parent class.
+        """
+        raise NotImplementedError
+
+    def _generate_statements(
+        self,
+        text: str,
+    ) -> list[str]:
+        """
+        Generate a list of statements from a piece of text, using a call to the LLM API.
+
+        Parameters
+        ----------
+        text: str
+            The text to extract statements from.
+
+        Returns
+        -------
+        list[str]
+            The list of statements extracted from the text.
+        """
+        messages = [
+            {"role": "system", "content": DEFAULT_SYSTEM_PROMPT},
+            {
+                "role": "user",
+                "content": _generate_statements_instruction(text),
+            },
+        ]
+
+        response = self(messages)
+        try:
+            response = trim_and_load_json(response)
+            assert type(response) == dict
+            assert "statements" in response
+            statements = response["statements"]
+        except Exception:
+            raise ValueError(
+                f"LLM response was not a valid list of statements: {response}"
+            )
+        return statements
+
+    def _generate_verdicts(
+        self,
+        query: str,
+        statements: list[str],
+    ) -> list:
+        """
+        Generates a list of answer relevance verdicts for a list of statements, using a call to the LLM API.
+
+        Parameters
+        ----------
+        query: str
+            The query to evaluate the statements against.
+        statements: list[str]
+            The statements to evaluate the validity of.
+
+        Returns
+        -------
+        list[dict[str,str]]
+            The list of verdicts for each statement. Each verdict is a dictionary with the "verdict" and optionally a "reason".
+        """
+        messages = [
+            {"role": "system", "content": DEFAULT_SYSTEM_PROMPT},
+            {
+                "role": "user",
+                "content": _generate_verdicts_instruction(query, statements),
+            },
+        ]
+
+        response = self(messages)
+
+        try:
+            response = trim_and_load_json(response)
+            assert type(response) == dict
+            assert "verdicts" in response
+            verdicts = response["verdicts"]
+            assert type(verdicts) == list
+            assert len(verdicts) == len(statements)
+            assert all(
+                verdict["verdict"] in ["yes", "no", "idk"]
+                for verdict in verdicts
+            )
+        except Exception:
```

Comment:
This looks messy and doesn't indicate to the user what the actual issue was with response. Also, the use of assert statements makes it unclear which fail conditions are actually being tested.